rssparser/src/main/java/com/prof/rssparser/core/parser/rss/RssParser.kt

```diff
@@ -263,7 +264,19 @@ internal fun extractRSSContent(xmlPullParser: XmlPullParser): Channel {
                 //region Mixed tags
                 xmlPullParser.contains(RSSKeyword.Image) -> when {
                     insideChannel && !insideItem -> insideChannelImage = true
-                    insideItem -> articleBuilder.image(xmlPullParser.nextTrimmedText())
+                    insideItem -> {
+                        try {
+                            // It would be ideal if we could peek ahead to see what is coming.
+                            // Attempt to get the image text if it's not contained in another tag
+                            articleBuilder.image(xmlPullParser.nextTrimmedText())
+                        } catch (exception: XmlPullParserException) {
+                            // if we get an exception from grabbing the next text,
+                            // assume we are inside the image tag and look inside the corresponding link tag and consume that
+                            if(xmlPullParser.contains(RSSKeyword.Link)) {
```

Comment:
definitely am not in love with this strategy - definitely open to other solutions!

Reply:
hmm, maybe I'm not following entirely but I believe the link provided covers a different scenario than what this is dealing with. There aren't any attributes to grab on the image or link tag in this example to probe. 

In the scenario above, the RSS we want to parse looks something like

```
<image>      <link>https://media.bleacherreport.com/image/upload/x_0,y_43,w_1800,h_1200,c_crop/v1685197298/ieh5bpyilj43f4dzpgua.jpg</link>
</image>
```

I believe that would work if the RSS looked like

```
<image link="https://media.bleacherreport.com/image/upload/x_0,y_43,w_1800,h_1200,c_crop/v1685197298/ieh5bpyilj43f4dzpgua.jpg"/>
```

Please let me know if I'm misunderstanding the approach you suggested!

Reply:
Ah right 🤔 

Alright, let's keep this logic for now and I will check if the Parser API provides a cleaner way.

Reply:
This seems to work! 

```kotlin
  xmlPullParser.next()
  val text = xmlPullParser.text.trim()
  // Get the image text if it's not contained in another tag
  if (text.isNotEmpty()) {
      articleBuilder.image(text)
  } else {
      xmlPullParser.next()
      if(xmlPullParser.contains(RSSKeyword.Link)) {
          articleBuilder.image(xmlPullParser.nextTrimmedText())
      }
  }
```

Weirdly, the call to `xmlPullParser.next()` always return the `EventType` `TEXT` even though there's a `link`. But if there's a `link` the current text is empty.